frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts

```diff
@@ -893,6 +893,17 @@ LIMIT 1000000
             },
         ],
 
+        isLikelyPastTTL: [
+            (s) => [s.start, s.snapshotSources],
+            (start, snapshotSources) => {
+                // If the recording is older than 30 days and has only realtime sources being reported, it is likely past its TTL
+                const isOlderThan30Days = dayjs().diff(start, 'hour') > 30
+                const onlyHasRealTime = snapshotSources?.every((s) => s.source === SnapshotSourceType.realtime)
+                const hasNoSources = snapshotSources?.length === 0
+                return isOlderThan30Days && (onlyHasRealTime || hasNoSources)
```

Comment:
we could specifically look this up when loading the recording
but it's cheaper to approximate it here 
this shouldn't ever match except when someone has directly loaded a stored URL for an old recording